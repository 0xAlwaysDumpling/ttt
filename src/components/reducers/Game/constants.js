export const boardSize = 3;
export const initialBoardState = new Array(boardSize).fill(null).map(row => new Array(boardSize).fill(null));

export const initialScoreboard = {
  p1w: 0,
  p2w: 0,
  p1l: 0,
  p2l: 0,
  draws: 0
}

export const initalGameHistoryMapState = (new Array(boardSize).fill(0)).reduce((accum, _, index) => { accum[index] = 0; return accum; }, {});
export const initialGameHistory = {
  hasEnded: false,
  playerturn: 0,
  winner: -1,
  row: { 0: { ...initalGameHistoryMapState }, 1: { ...initalGameHistoryMapState } },
  col: { 0: { ...initalGameHistoryMapState }, 1: { ...initalGameHistoryMapState } },
  diag: { 0: 0, 1: 0 },
  antidiag: { 0: 0, 1: 0 },
}

export const initialState = {
  boardState: initialBoardState,
  gameHistory: initialGameHistory,
  scoreboard: initialScoreboard,
};

export const resetGameState = {
  boardState: initialBoardState,
  gameHistory: initialGameHistory
};
