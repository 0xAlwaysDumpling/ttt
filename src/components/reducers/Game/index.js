
import { loadState } from '../../localstorage';
import { initialState, resetGameState } from './constants';
import { postMove } from '../../api';

/* 
  To mock API call I needed to do actions MOVE_REQUEST, MOVE_SUCCESS, MOVE_FAILED 
  In the interest of scope and time I just made a function call postMove

*/
const GameState = (state = initialState, action) => {
  switch (action.type) {
    case 'LOAD':
      const appState = loadState();
      return appState ? appState : initialState;
    case 'MOVE':
      if (state.gameHistory.hasEnded) return state;
      else return postMove(state, action)
    case 'RESET':
      return {
        ...state,
        ...resetGameState
      };
    default:
      return state;
  }
}

export default GameState;