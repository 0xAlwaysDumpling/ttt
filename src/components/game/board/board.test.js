import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Board } from './index';
import { initialBoardState } from '../../reducers/Game/constants';

afterEach(cleanup)

const onMove = () => { };
it('should render with out crashing', () => {

  render(<Board boardState={initialBoardState} onMove={onMove} />)
});

it('should take a snapshot of board', () => {
  const { asFragment } = render(<Board boardState={initialBoardState} onMove={onMove} />)

  expect(asFragment(<Board boardState={initialBoardState} onMove={onMove} />)).toMatchSnapshot()
});
