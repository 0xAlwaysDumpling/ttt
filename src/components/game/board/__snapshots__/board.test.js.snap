// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of board 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ hJmkar"
  >
    <div
      class="sc-AxiKw CYfUp"
    >
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
    </div>
    <div
      class="sc-AxiKw CYfUp"
    >
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
    </div>
    <div
      class="sc-AxiKw CYfUp"
    >
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
      <div
        class="sc-AxhCb exkQbL"
      >
        <button
          class="sc-AxjAm dRYkuE"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
